src/Transports/MassTransit.KafkaIntegration/Filters/KafkaCreateTopicFilter.cs

```diff
@@ -0,0 +1,71 @@
+namespace MassTransit.KafkaIntegration.Filters
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Confluent.Kafka;
+    using Confluent.Kafka.Admin;
+    using Context;
+    using GreenPipes;
+    using Logging;
+    using Transport;
+
+
+    public class KafkaCreateTopicFilter<TKey, TValue> :
+        IFilter<ConsumerContext<TKey, TValue>>
+        where TValue : class
+    {
+        readonly AdminClientConfig _config;
+        readonly ReceiveEndpointContext _context;
+        readonly CreateTopicsOptions _options;
+        readonly IEnumerable<TopicSpecification> _specifications;
+        bool _hasBeenCalled;
+        bool _hasBeenCreated;
+
+        public KafkaCreateTopicFilter(ReceiveEndpointContext context, IReadOnlyDictionary<string, string> clientConfig,
+            IEnumerable<TopicSpecification> specifications)
+        {
+            _context = context;
+            _specifications = specifications;
+            _options = new CreateTopicsOptions();
+            _config = new AdminClientConfig(clientConfig.ToDictionary(x => x.Key, x => x.Value));
+        }
+
+        public async Task Send(ConsumerContext<TKey, TValue> context, IPipe<ConsumerContext<TKey, TValue>> next)
+        {
+            if (_hasBeenCreated)
```

Comment:
hasBeenCreated? hasBeenCalled? confusing, vs use the OneTimeStep like the ConfigureTopologyFilter